client/src/components/carousel.jsx

```diff
@@ -19,28 +20,44 @@ class Carousel extends React.Component {
     this.state = {
       images: dummy,
       scaled: 0,
-      isFavorite: false //there is no way to keep track of users right now.
+      isFavorite: false, //there is no way to keep track of users right now.
+      modalActive: false
     };
 
     this.retrieveImageDocument = this.retrieveImageDocument.bind(this);
     this.changeFavorite = this.changeFavorite.bind(this);
     this.cycleForward = this.cycleForward.bind(this);
     this.cycleBack = this.cycleBack.bind(this);
     this.handleThumbnailClick = this.handleThumbnailClick.bind(this);
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+
   }
 
   //******************** EVENT HANDLERS ********************
   changeFavorite() {
     this.setState({
       isFavorite: !this.state.isFavorite
+    }, () => {
+      console.log(`Produc favorite is now: ${this.state.isFavorite ? 'yes' : 'no'}`);
+    });
+  }
+
+  openModal() { //it will probably be best to have a separate function for hide and show
+    this.setState({
+      modalActive: true
     });
-    //TODO: need to visually show the change, for now just log
-    console.log(`Is this product favorited? ${this.state.isFavorite ? 'yes' : 'no'}`);
   }
 
-  handleThumbnailClick(val) {
+  closeModal() {
     this.setState({
-      scaled: val
+      modalActive: false
+    });
+  }
```

Comment:
you could also just create one function called something like `toggleModal` that just flips the state from what it currently is

Reply:
I did not do that because it's really easy for browsers/mice/trackpads to register multiple clicks for what should be a single click, which could potentially swap it twice and leave it open